backend/tests/e2e_tests/__init__.py

```diff
@@ -37,16 +40,32 @@
 }
 
 
+def default_log_fn(*args, **kwargs):
+    print(*args, **kwargs)
+
+
 @dataclass
 class UserClient:
     name: str
     backend: str
     token: str
     payment_method: str
+    log_fn: typing.Callable
 
     @staticmethod
-    def create(backend: str, name: str) -> "UserClient":
-        return create_test_user(backend, f"{name}_{random.randint(0, 1000)}")
+    def create(
+        backend: str,
+        name: str,
+        kyc_info=None,
+        log_fn=default_log_fn,
+        randomize=True,
+    ) -> "UserClient":
+        return create_test_user(
+            backend,
+            f"{name}_{random.randint(0, 10000000)}" if randomize else name,
```

Comment:
I think it is probably easier and simpler to just have global user id int, and you increase 1 every time create a new user, which ensures we have uniq id.
then you don't need special unique_username func.
I don't see any reason we don't want a uniq name in test. hence less option is better, always generate a uniq name.